src/content_scripts/applyColorSchemeJs.js

```diff
@@ -136,15 +311,54 @@ function matchMediaOverwrite(...args) {
 function applyJsOverwrite() {
     // do not overwrite twice
     if (overwroteMatchMedia) {
+        if (jsColorStatus !== fakedColorStatus) {
+            dispatchChangeEvents();
+        }
         return;
     }
 
-    // add hint for websites to detect this has been faked
-    // matchMediaOverwrite.FAKED = ADDON_FAKED_WARNING;
-    // DISABLED, as it could be used for tracking (which simple console statements itself cannot AFAIK)
+    // actually overwrite
+
+    Reflect.defineProperty(MediaQueryListPrototype, 'addListener', {
+        configurable: true,
+        enumerable: true,
+        value: exportFunction(skeleton.addListener, window),
+        writable: true
+    });
+    Reflect.defineProperty(MediaQueryListPrototype, 'removeListener', {
+        configurable: true,
+        enumerable: true,
+        value: exportFunction(skeleton.removeListener, window),
+        writable: true
+    });
+    let descMatches = Reflect.getOwnPropertyDescriptor(skeleton, 'matches');
```

Comment:
Better use variable names without abbreviations to make it easier to understand/and less easier to mistake. I would translate `descMatches` as `descriptionMatches` e.g., but I doubt that is correct.

Same below.

Reply:
It's "descriptor".